### PR TITLE
feat: add search c# template

### PR DESCRIPTION
**Pull request by yukun-dong**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/24719436

### CODE REVIEW

**Comment by swatDong** [comment]
Are you going to own this template? It seems we are added as owners due to the wildcard "*/templates/\*/m365-message-extension*". Feel free to update CODEOWNERS to separate node and c#.

**Comment by JerryYangKai** [comment]
> Are you going to own this template? It seems we are added as owners due to the wildcard "_/templates/*/m365-message-extension_". Feel free to update CODEOWNERS to separate node and c#.

Agreed, the folder name may cause some misunderstanding..

**Comment by yukun-dong** [comment]
> Are you going to own this template? It seems we are added as owners due to the wildcard "_/templates/*/m365-message-extension_". Feel free to update CODEOWNERS to separate node and c#.

Thanks for reminding. Let me update later.

File: templates/csharp/m365-message-extension/Search/SearchApp.cs.tpl [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                Attachments = attachments
+            }
+        };
+    }
+
+    // Generate a set of substrings to illustrate the idea of a set of results coming back from a query. 
+    private async Task<IEnumerable<(string, string, string, string, string)>> FindPackages(string text)
+    {
+        var obj = JObject.Parse(await (new HttpClient()).GetStringAsync($"https://azuresearch-usnc.nuget.org/query?q=id:{text}&prerelease=true"));
+        return obj["data"].Select(item => (item["id"].ToString(), item["version"].ToString(), item["description"].ToString(), item["projectUrl"]?.ToString(), item["iconUrl"]?.ToString()));
```
Comment (JerryYangKai) [inline templates/csharp/m365-message-extension/Search/SearchApp.cs.tpl:75]: This line seems unreadable. 
I know Teams Samples also implement it in this way.
Maybe we can just change some style, like make all the item[X].ToString() lines paralleled.

**Review by JerryYangKai — APPROVED** [review]

**Pull request merged**